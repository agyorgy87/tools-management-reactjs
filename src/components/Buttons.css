@import url('https://fonts.googleapis.com/css2?family=Jura&display=swap');

.Filter {
    text-align: left;
    color:#fe8e22;
    font-family: 'Jura', sans-serif;
    font-size: 30px;
}

.MainButtons {
    margin-top: 20px;
    display: flex;
    justify-content: space-around;
}

.ButtonSet {
    height: 30px;
    border: 1px solid white;
    border-radius: 5px;
    background-color: #41414d;
    color: white;
    font-family: 'Jura', sans-serif;
}

.ButtonSet:hover {
    color: #fe8e22;
}
