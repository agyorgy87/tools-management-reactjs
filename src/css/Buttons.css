@import url('https://fonts.googleapis.com/css2?family=Jura&display=swap');

.filter-text{ 
    text-align: left;
    color:#fe8e22;
    font-family: 'Jura', sans-serif;
    font-size: 30px;
}

.filter-buttons-container {
    margin-top: 20px;
    display: flex;
    justify-content: space-around;
}

.filter-buttons {
    height: 30px;
    border: 1px solid white;
    border-radius: 5px;
    background-color: #41414d;
    color: white;
    font-family: 'Jura', sans-serif;
}

.filter-buttons:hover {
    color: #fe8e22;
}
