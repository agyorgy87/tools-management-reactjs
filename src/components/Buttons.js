import React, {useState} from 'react';
import '../css/Buttons.css';

export default function Buttons (props) {

    let tools = props.data;
    const[allData, setAllData] = useState(tools);

    const[filterValue, setFilterValue] = useState([])

    const AllTools = (props) => {
        props.setData(allData);
    }

    const Available = (props) => {
        setFilterValue([]);
        const allDataForAvailableItems = allData;
        const availableDataResult = allDataForAvailableItems.filter(item => item.Quantity !== 0);
        console.log(availableDataResult);
        filterValue.push(...availableDataResult);
        props.setData(filterValue)
    }

    const Cheapest = (props) => {
        setFilterValue([]);
        const allDataForCheapestValue = allData;
        const theMinPriceOfAllData = allDataForCheapestValue.map(element => element.Price);
        let minValue = Math.min( ...theMinPriceOfAllData);
        let CheapestData = allDataForCheapestValue.filter(item => item.Price === minValue);
        filterValue.push(...CheapestData);
        props.setData(filterValue);
    }

     const MostExpensive  = (props) => {
        setFilterValue([]);
        const allDataForCheapestValue = allData;
        const theMaxPriceOfAllData = allDataForCheapestValue.map(element => element.Price);
        let maxValue = Math.max( ...theMaxPriceOfAllData );
        let MostExpensiveData = allDataForCheapestValue.filter(item => item.Price === maxValue);
        filterValue.push(...MostExpensiveData);
        props.setData(filterValue);
    }

    return(
        <div>
            <header>
                <h3 className="filter-text">Filter</h3>               
            </header>
            <main className="filter-buttons-container">
                <button onClick={() => AllTools(props)} className="filter-buttons">ALL TOOLS</button>
                <button onClick={() => Available(props)} className="filter-buttons">CURRENTLY IN STOCK</button>
                <button onClick={() => Cheapest(props)} className="filter-buttons">CHEAPEST</button>
                <button onClick={() => MostExpensive(props)} className="filter-buttons">MOST EXPENSIVE</button>  
            </main>
        </div>
    )
}