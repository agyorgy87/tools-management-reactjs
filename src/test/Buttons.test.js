import renderer from 'react-test-renderer';
import Buttons from "../components/Buttons.js";

test("test Buttons", () => {
    let tree = renderer
    .create(<Buttons data={[]} setData={() => {}}/>)
    .toJSON()
    expect(tree).toMatchSnapshot();
});