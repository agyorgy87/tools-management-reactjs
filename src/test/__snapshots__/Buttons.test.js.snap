// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Buttons 1`] = `
<div>
  <header>
    <h3
      className="filter-text"
    >
      Filter
    </h3>
  </header>
  <main
    className="filter-buttons-container"
  >
    <button
      className="filter-buttons"
      onClick={[Function]}
    >
      ALL TOOLS
    </button>
    <button
      className="filter-buttons"
      onClick={[Function]}
    >
      CURRENTLY IN STOCK
    </button>
    <button
      className="filter-buttons"
      onClick={[Function]}
    >
      CHEAPEST
    </button>
    <button
      className="filter-buttons"
      onClick={[Function]}
    >
      MOST EXPENSIVE
    </button>
  </main>
</div>
`;
